cli/command_repository_upgrade.go

```diff
@@ -0,0 +1,291 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/alecthomas/kingpin"
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/internal/epoch"
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/content"
+)
+
+type commandRepositoryUpgrade struct {
+	forceRollback bool
+	skip          bool
+	force         bool
+
+	// lock settings
+	advanceNoticeDuration time.Duration
+	ioDrainTimeout        time.Duration
+	statusPollInterval    time.Duration
+
+	svc advancedAppServices
+}
+
+const (
+	experimentalWarning = `WARNING: The upgrade command is an EXPERIMENTAL feature. Please use with caution.
```

Comment:
```suggestion
	experimentalWarning = `WARNING: The upgrade command is an EXPERIMENTAL feature. Please DO NOT use it, it may corrupt your repository and cause data loss.
```